src/PruneUnusedContainersModal.jsx

```diff
@@ -0,0 +1,110 @@
+            sortKey: container.system.toString(),
+            props: {
+                className: "ignore-pixels",
+                modifier: "nowrap"
+            }
+        });
+
+    return { columns, selected, props: { key: container.id } };
+};
+
+const PruneUnusedContainersModal = ({ close, unusedContainers, onAddNotification, userSystemServiceAvailable, user }) => {
+    const [isPruning, setPruning] = useState(false);
+    const [selectedContainerIds, setSelectedContainerIds] = React.useState(unusedContainers.map(u => u.id));
+
+    const handlePruneUnusedContainers = () => {
+        setPruning(true);
+
+        const actions = [];
+
+        for (const id of selectedContainerIds) {
+            if (id.endsWith("true")) {
+                actions.push(client.delContainer(true, id.replace(/true$/, ""), true));
+            } else {
+                actions.push(client.delContainer(false, id.replace(/false$/, ""), true));
+            }
+        }
+        Promise.all(actions).then(close)
+                .catch(ex => {
+                    const error = _("Failed to prune unused containers");
+                    onAddNotification({ type: 'danger', error, errorDetail: ex.message });
+                    close();
+                });
+    };
+
+    const columns = [
+        { title: _("Name"), sortable: true },
+        { title: _("Created"), sortable: true },
+    ];
+
+    if (userSystemServiceAvailable)
+        columns.push({ title: _("Owner"), sortable: true });
+
+    const selectAllContainers = isSelecting => setSelectedContainerIds(isSelecting ? unusedContainers.map(c => c.id) : []);
+    const isContainerSelected = container => selectedContainerIds.includes(container.id);
+    const setContainerSelected = (container, isSelecting) => setSelectedContainerIds(prevSelected => {
+        const otherSelectedContainerNames = prevSelected.filter(r => r !== container.id);
+        return isSelecting ? [...otherSelectedContainerNames, container.id] : otherSelectedContainerNames;
+    });
+
+    const onSelectContainer = (id, _rowIndex, isSelecting) => {
+        const container = unusedContainers.filter(u => u.id === id)[0];
+        setContainerSelected(container, isSelecting);
+    };
+
+    return (
+        <Modal isOpen
+               onClose={close}
+               position="top" variant="medium"
+               title={cockpit.format(_("Prune unused containers"))}
+               footer={<>
+                   <Button id="btn-img-delete" variant="danger"
+                           spinnerAriaValueText={isPruning ? _("Pruning containers") : undefined}
+                           isLoading={isPruning}
+                           isDisabled={selectedContainerIds.length === 0}
+                           onClick={handlePruneUnusedContainers}>
+                       {isPruning ? _("Pruning containers") : _("Prune")}
```

Comment:
Changing the string to something so much longer will make the Cancel button jump to the right.

Additionally I searched quickly and we don't change the string of buttons on 'loading' elsewhere.  So I suggest we keeo the original string intact.

Reply:
Hmm our prune unused images modal does this as well, so I guess I could change it to `Pruning`.

Reply:
Actually `Pruning` also changes size hmm.